src/main/java/ure/URECaveGenerator.java

```diff
@@ -0,0 +1,41 @@
+package ure;
+
+import java.lang.Math;
+
+public class URECaveGenerator extends UREAreaGenerator {
+    @Override
+    public UREArea generate(URETerrainCzar terrainCzar, int totalSteps, int sizeX, int sizeY) {
+        UREArea area = new UREArea(sizeX, sizeY, terrainCzar.getTerrainForFilechar('W'));
```

Comment:
as i am sure you realized, it would be better to pass in 'use this char for floors and this char for walls' etc.  which means this method can't really be generic.  but that's probably fine.